### ISSUE TITLE
Umbrella: Powder diffraction instruments migration to PEARL scripts

### ISSUE THREAD

**Issue by DavidFair** [mantidproject/mantid#17831]
This umbrella issue holds the high level steps needed for the migration of powder diffraction instruments at ISIS to the PEARL powder diffraction script.

**Note this ticket will update and change as items are added to the roadmap**

### Completed:
 Prep work:
~~#17839~~
### Refactoring Work:
~~#18108~~

---
## All Instruments
#19113 - Unit tests
#19105 - Switch to `AlignAndFocusPowder`
- [x] Update release notes when we are happy with the scripts. 
- [x] Ensure system tests tidy up after themselves
- [x] Abstract instrument should have a default behaviour which instruments then override - e.g. `NormaliseByCurrent` this should help reduce the amount of boilerplate written in each instrument class.
- [x]  Create an example calibration folder for each instrument that shows the file structure
- [x] Include the mandatory parameters for each method in the documentation 
- [x] Ensure everything in `isis_powder` is Python 3 ready - see #18550 

# POLARIS and PEARL

- [x] Add option to specify ext (single file only) for loading intermediate files

### PEARL:
- [x] Add method for creating .cal file at a later date
- [x] We should move the remaining hardcoded values into the adv. config file

### POLARIS:
~~#17949~~
- [x] As part of above at a later date focused runs should also be grouped possible 

# IMAT & GEM
#18645 - IMAT and ENGIN-X
#18643 - GEM
## IMAT
- [ ] Refactor the Engineering Diffraction GUI to include IMAT

# Low priority
- [ ] Give the option to use a flat directory for the calibration directory - POLARIS
- [x] Some scientists require GEM to have its banks further subdivided according to a grouping file. This should be trivial to add to GEM once everything else is in place
- [x] Various workspaces leak out of the script - once all work is completed we should identify where we are not deleting workspaces (usually because the label in Python gets reassigned to a new workspace)
- [x] Paths should expand path macros when encountered
- [ ] Possibly look into detecting when a run has been completed and processing that. Data Reduction Service?

**Comment by DavidFair** [comment]
As this is an ongoing process moving the ticket to 3.10

**Comment by Anders-Markvardsen** [comment]
From discussion with @josephframsay and @DanNixon this issue is out-of-date and requirements and priorities have changed since this issue was last worked on
